packages/components/src/toggle-group-control/stories/index.tsx

```diff
@@ -31,6 +35,7 @@ const meta: ComponentMeta< typeof ToggleGroupControl > = {
 		help: { control: { type: 'text' } },
 		onChange: { action: 'onChange' },
 		value: { control: { type: null } },
+		defaultValue: { control: { type: null } },
```

Comment:
I was thinking about why we still need to add a `defaultValue` prop when we're supporting `undefined` values, and then realized we have another breaking change here 😭

On trunk, we used to be able to just give a "default value" to the `value` prop and use the component as uncontrolled:

```jsx
		<ToggleGroupControl
			{ ...props }
			onChange={ ( ...changeArgs ) => {
				onChange?.( ...changeArgs );
			} }
			value={ 'left' }
		/>
```

Could we maybe change the `hasEverBeenUsedInControlledMode` heuristics to see if the `value` has changed at least once? (Kind of seems like `usePrevious` might be inevitable 🤔)

Reply:
Alright, I gave this approach a try in https://github.com/WordPress/gutenberg/pull/50278/commits/288eb0b14735ad54880a85ddd80f9fdf29c788a7

Let me know if it now looks good to you!